source/common/ssl/ssl_socket.cc

```diff
@@ -227,6 +228,30 @@ std::string SslSocket::sha256PeerCertificateDigest() {
   return Hex::encode(computed_hash);
 }
 
+std::string SslSocket::urlEncodedPemEncodedPeerCertificate() const {
```

Comment:
As an aside, I think that for perf reasons we could cache the result of this inside this class. For kept alive connections, it's very likely this will end up getting called many times. I would just keep that in mind for the future if this shows up on perf traces. Can you add a TODO?

Reply:
Sure, I was planning to ask *how* to do this after I landed this first PR.  I agree that many of these values should be cached, but subjects/other things also weren't caching and I figured I might hit all of them in the same second PR.